core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -86,6 +86,19 @@ public void addActivity(Activity activity)
 		activity.setDayActivity(this);
 		activity.setActivityCategory(getGoal().getActivityCategory());
 		this.activities.add(activity);
+
+		this.resetAggregatesComputed();
+	}
+
+	@Override
+	protected void resetAggregatesComputed()
+	{
+		super.resetAggregatesComputed();
+
+		if (weekActivity != null)
```

Comment:
Yes. Scattering the code with null checks is a common pattern, but it reduces readability. Let's only put them in when they're really needed. The same comment also occurs in ``Activity``.